apps/vmq_server/src/vmq_ranch_config.erl

```diff
@@ -334,18 +334,24 @@ protocol_opts(cowboy_clear, Type, Opts) when
         stream_handlers => [vmq_cowboy_websocket_h, cowboy_stream_h]
     });
 protocol_opts(cowboy_clear, _, Opts) ->
-    Routes =
-        case {lists:keyfind(config_mod, 1, Opts), lists:keyfind(config_fun, 1, Opts)} of
-            {{_, ConfigMod}, {_, ConfigFun}} ->
-                try
-                    apply(ConfigMod, ConfigFun, [])
-                catch
-                    E:R ->
-                        lager:error("can't setup HTTP modules due to ~p:~p", [E, R]),
-                        []
-                end;
-            _ ->
+    Apply = fun(Mod, Fun, Param) ->
+        try
+            apply(Mod, Fun, Param)
+        catch
+            E:R ->
+                lager:error("can't setup HTTP modules due to ~p:~p", [E, R]),
                 []
+        end
+    end,
+    {_, ConfigMod} = lists:keyfind(config_mod, 1, Opts),
```

Comment:
This implies that we always have a `config_mod` and `config_fun` in Opts, otherwise we crash here. I guess that's the case via defaults, though.

Reply:
Following the least-surprise-principle: Not configuring any (http/https) listener...

Reply:
You're right, I don't see any need to compile empty routes, which is technically possible. I don't know why we originally allowed it.

```
(VerneMQ@127.0.0.1)9> cowboy_router:compile([{'_', []}]).
[{'_',[],[]}]
```
Let's keep your version.